ServerCore/Pages/Swag/TeamReport.cshtml.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Swag
+{
+    /// <summary>
+    /// Model for the admin's view of the swag items for a team
+    /// </summary>
+    [Authorize(Policy = "IsEventAdmin")]
+    public class TeamReportModel : EventSpecificPageModel
+    {
+        public TeamReportModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        /// <summary>
+        /// Individual order for a team
+        /// </summary>
+        public class SwagView
+        {
+            public string ContactEmail { get; set; }
+            public string TeamName { get; set; }
+            public string Lunch { get; set; }
+            public int TeamId { get; set; }
+        }
+
+        /// <summary>
+        /// Aggregate orders for an item
+        /// </summary>
+        public class SummaryItem
+        {
+            public string Lunch { get; set; }
+            public int Count { get; set; }
+        }
+
+        public List<SwagView> SwagViews { get; set; }
+        public List<SummaryItem> Summary { get; set; }
+
+        public async Task<IActionResult> OnGetAsync()
+        {
+            // Fetch lunches the team chose
+            SwagViews = await (from teamLunch in _context.TeamLunch
+                         join team in _context.Teams on teamLunch.TeamId equals team.ID
```

Comment:
question: Does this need a join if Team is already a virtual navigation property of TeamLunch?

Reply:
Using the navigation property also does a join, this is just more explicit